src/CoreWCF.WebHttp/src/CoreWCF/UriTemplateTable.cs

```diff
@@ -0,0 +1,616 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.Diagnostics;
+using CoreWCF.Runtime;
+
+namespace CoreWCF
+{
+    public class UriTemplateTable
+    {
+        private Uri _baseAddress;
+        private string _basePath;
+        private Dictionary<string, FastPathInfo> _fastPathTable; // key is uri.PathAndQuery, fastPathTable may be null
+        private bool _noTemplateHasQueryPart;
+        private int _numSegmentsInBaseAddress;
+        private Uri _originalUncanonicalizedBaseAddress;
+        private UriTemplateTrieNode rootNode;
+        private readonly UriTemplatesCollection _templates;
+        private readonly object _thisLock;
+        private readonly bool _addTrailingSlashToBaseAddress;
+
+        public UriTemplateTable()
+            : this(null, null, true)
+        {
+        }
+        public UriTemplateTable(IEnumerable<KeyValuePair<UriTemplate, object>> keyValuePairs)
+            : this(null, keyValuePairs, true)
```

Comment:
Spacing between methods, multiple places in this file

Reply:
Addressed in 14ac88e67a220d805d2fac57fa31026cd397ba1e.